### PR TITLE
Initial work compression benchmark

### PR DESCRIPTION
**Pull request by fylux**
I am trying to develop a benchmark to measure the compression ratio of different algorithms. This works so far but I am still looking for how to measure file sizes with GoogleBenchmarks, since right now it only measures speeds.

Also, I am using RDF initial data to test the algorithms.

### CODE REVIEW

File: root/io/io/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,9 @@
 RB_ADD_GBENCHMARK(IOBenchmarks
   TBufferMergerBenchmarks.cxx
   LABEL short
-  LIBRARIES Core RIO Tree TreePlayer)
\ No newline at end of file
+  LIBRARIES Core RIO Tree TreePlayer)
+
+RB_ADD_GBENCHMARK(CompressBenchmarks
```
Comment (oshadura) [inline root/io/io/CMakeLists.txt]: @fylux, thank you for your initial work on compression benchmark!

File: root/io/io/TFileBenchmarks.cxx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+static void BM_TFile_Write(benchmark::State &state) {
+	ROOT::EnableImplicitMT();
+	ROOT::RDF::RSnapshotOptions options;
+	options.fCompressionAlgorithm = ROOT::ECompressionAlgorithm::kZLIB;
+	//RCompressionSetting::EAlgorithm::EValues::kZLIB
+
+	for (auto _ : state) {
+		// We create an empty data frame
+	   ROOT::RDataFrame tdf(100000);
```
Comment (oshadura) [inline root/io/io/TFileBenchmarks.cxx]: Here you are benchmarking too many extra things, such as building RDataframe and filling it with random numbers, can we try to use fixture? What you actually need to benchmark in this case is Snapshot function https://root.cern/doc/master/classROOT_1_1RDF_1_1RInterface.html#a233b7723e498967f4340705d2c4db7f8.

File: root/io/io/TFileBenchmarks.cxx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#include "ROOT/RSnapshotOptions.hxx"
+
+#include "benchmark/benchmark.h"
+#include "rootbench/RBConfig.h"
+
+
+static void BM_TFile_Write(benchmark::State &state) {
+	ROOT::EnableImplicitMT();
+	ROOT::RDF::RSnapshotOptions options;
+	options.fCompressionAlgorithm = ROOT::ECompressionAlgorithm::kZLIB;
```
Comment (oshadura) [inline root/io/io/TFileBenchmarks.cxx]: Add please tests also for other compression algorithms: LZMA, LZ4.

File: root/io/io/TFileBenchmarks.cxx (hunk 2 above)
Comment (fylux) [reply root/io/io/TFileBenchmarks.cxx]: Yes, I forgot to move them out of the loop. What do you mean with fixture?

File: root/io/io/TFileBenchmarks.cxx (hunk 3 above)
Comment (fylux) [reply root/io/io/TFileBenchmarks.cxx]: Added

File: root/io/io/TFileBenchmarks.cxx (hunk 2 above)
Comment (oshadura) [reply root/io/io/TFileBenchmarks.cxx]: Check about fixtures: https://github.com/google/benchmark#fixtures

File: root/io/io/TFileBenchmarks.cxx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+	for (auto _ : state) {
+	   	//And we write out the dataset on disk
+	   	tdf_1.Snapshot("randomNumbers", "bench_data.root", "", options);
+	}
+}
+BENCHMARK(BM_TFile_Write_LZ4)->Unit(benchmark::kMicrosecond);
+
+
+static void BM_TFile_Write_kLZMA (benchmark::State &state) {
```
Comment (oshadura) [inline root/io/io/TFileBenchmarks.cxx]: kLZMA->LZMA(if to be more consistent with other names)

File: root/io/io/TFileBenchmarks.cxx [hunk 5]
```diff
@@ -0,0 +1,72 @@
+#include "TROOT.h"   // ROOT::EnableImplicitMT
+
+#include "TFile.h"	
+#include "TFrame.h"
+#include "TRandom3.h"
+#include "TBenchmark.h"
```
Comment (oshadura) [inline root/io/io/TFileBenchmarks.cxx]: Do we really need this include?

File: root/io/io/TFileBenchmarks.cxx [hunk 6]
```diff
@@ -0,0 +1,72 @@
+#include "TROOT.h"   // ROOT::EnableImplicitMT
+
+#include "TFile.h"	
+#include "TFrame.h"
```
Comment (oshadura) [inline root/io/io/TFileBenchmarks.cxx]: Do we really need this include?

File: root/io/io/TFileBenchmarks.cxx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+#include "TBenchmark.h"
+
+#include "ROOT/RDataFrame.hxx"
+#include "ROOT/RSnapshotOptions.hxx"
+
+#include "benchmark/benchmark.h"
+#include "rootbench/RBConfig.h"
+
+
+static void BM_TFile_Write_ZLIB(benchmark::State &state) {
```
Comment (oshadura) [inline root/io/io/TFileBenchmarks.cxx]: Can you rename please "BM_TFile_Write_ZLIB" to something more explicitly saying that here we are testing Snapshot action? Maybe "BM_TFile_RDFSnapshot_ZLIB" Thanks!

File: root/io/io/TFileBenchmarks.cxx (hunk 2 above)
Comment (fylux) [reply root/io/io/TFileBenchmarks.cxx]: I am thinking how to apply it to this particular case. The example are a bit confusing if you are not familiar with the framework.
Should it be something like the following?

```
class MyFixture : public ::benchmark::Fixture {
public:
    void SetUp(const ::benchmark::State& st) {
       ROOT::EnableImplicitMT();
	   ROOT::RDF::RSnapshotOptions options;
	   options.fCompressionAlgorithm = ROOT::ECompressionAlgorithm::kLZ4;

	   // We create an empty data frame
	   ROOT::RDataFrame tdf(100000);
	   // We now fill it with random numbers
	   gRandom->SetSeed(1);
	   auto tdf_1 = tdf.Define("rnd", []() { return gRandom->Gaus(); });
	}

    void TearDown(const ::benchmark::State&) {}
    auto tdf_1;
};

BENCHMARK_F(MyFixture, BM_TFile_RDFSnapshot_ZLIB)(benchmark::State &state) {
	for (auto _ : state) {
		//And we write out the dataset on disk
		tdf_1.Snapshot("randomNumbers", "bench_data.root", "", options);
	}
}
```

**Comment by oshadura** [comment]
@fylux, can you please rebase on master? I see that branch is still out-of-date

**Comment by fylux** [comment]
@oshadura It should be ok now.

**Pull request merged**